frame/collective/src/lib.rs

```diff
@@ -1441,6 +1458,186 @@ mod tests {
 		});
 	}
 
+	#[test]
+	fn motions_all_first_vote_free_works() {
+		new_test_ext().execute_with(|| {
+			let proposal = make_proposal(42);
+			let proposal_len: u32 = proposal.using_encoded(|p| p.len() as u32);
+			let hash: H256 = proposal.blake2_256().into();
+			let end = 4;
+			assert_ok!(Collective::propose(Origin::signed(1), 2, Box::new(proposal.clone()), proposal_len));
+			assert_eq!(
+				Collective::voting(&hash),
+				Some(Votes { index: 0, threshold: 2, ayes: vec![1], nays: vec![], end })
+			);
+
+			// For the motion, acc 2's first vote, expecting Ok with Pays::No ...
+			assert_eq!( Collective::vote(Origin::signed(2), hash.clone(), 0, true),
+				Ok(
+					PostDispatchInfo {
+						actual_weight: Some(
+							207711000,
+						),
+						pays_fee: Pays::No,
+					}
+				)
+			);
+
+			// Duplicate vote, expecting error with Pays::Yes ...
+			let vote_rval: DispatchResultWithPostInfo = Collective::vote(Origin::signed(2), hash.clone(), 0, true);
+			match vote_rval {
+				Ok(_) => {
+					println!( "@[{:#?}::{:#?}]::vote-fee() | Should not Occur",
+						file!(),
+						line!() );
+				},
+				Err(err) => {
+					assert_eq!( err.post_info.pays_fee, Pays::Yes );
+				},
+			}
+
+			// Modifying vote, expecting ok with Pays::Yes ...
+			assert_eq!( Collective::vote(Origin::signed(2), hash.clone(), 0, false),
+				Ok(
+					PostDispatchInfo {
+						actual_weight: Some(
+							207711000,
+						),
+						pays_fee: Pays::Yes,
+					}
+				)
+			);
+
+			// For the motion, acc 3's first vote, expecting Ok with Pays::No ...
+			assert_eq!( Collective::vote(Origin::signed(3), hash.clone(), 0, true),
+				Ok(
+					PostDispatchInfo {
+						actual_weight: Some(
+							207711000,
+						),
+						pays_fee: Pays::No,
+					}
+				)
+			);
+
+			// acc 3 modify the vote, expecting Ok with Pays::Yes ...
+			assert_eq!( Collective::vote(Origin::signed(3), hash.clone(), 0, false),
+				Ok(
+					PostDispatchInfo {
+						actual_weight: Some(
+							207711000,
+						),
+						pays_fee: Pays::Yes,
+					}
+				)
+			);
+
+			// Test close() Extrincis | Check DispatchResultWithPostInfo with Pay Info
+
+			let proposal_weight = proposal.get_dispatch_info().weight;
+
+			// trying to close the valid proposal, Expecting OK with Pays::No
+			assert_eq!(
+				Collective::close(Origin::signed(2), hash.clone(), 0, proposal_weight, proposal_len),
+				Ok(
+					PostDispatchInfo {
+						actual_weight: Some(
+							437711000,
+						),
+						pays_fee: Pays::No,
+					},
+				)
+			);
+
+			// trying to close the proposal, which is already closed. Expecting error "ProposalMissing" with Pays::Yes
+			let close_rval: DispatchResultWithPostInfo = Collective::close(Origin::signed(2), hash.clone(), 0, proposal_weight, proposal_len);
+			match close_rval {
+				Ok(_) => {
+					println!( "@[{:#?}::{:#?}]::vote-fee() | Should not Occur",
+						file!(),
+						line!() );
```

Comment:
Should be another panic situation, forcing the test to fail

Reply:
Thanks Shawn, replaced with suggested one.

![](https://i.imgur.com/lfd2Bwo.png)